src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/Visitors/CSharpTagHelperAttributeValueVisitor.cs

```diff
@@ -0,0 +1,193 @@
+        /// Writes code for the given <paramref name="chunk"/>.
+        /// </summary>
+        /// <param name="chunk">The <see cref="SectionChunk"/> to render.</param>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown unconditionally to inform user of unexpected <c>@section</c> directive.
+        /// </exception>
+        protected override void Visit(SectionChunk chunk)
+        {
+            throw new InvalidOperationException(RazorResources.FormatTagHelpers_Directives_NotSupported_InAttributes(
+                SyntaxConstants.CSharp.SectionKeyword));
+        }
+
+        /// <summary>
+        /// Writes code for the given <paramref name="chunk"/>.
+        /// </summary>
+        /// <param name="chunk">The <see cref="SetLayoutChunk"/> to render.</param>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown unconditionally to inform user of unexpected <c>@layout</c> directive.
+        /// </exception>
+        protected override void Visit(SetLayoutChunk chunk)
+        {
+            throw new InvalidOperationException(RazorResources.FormatTagHelpers_Directives_NotSupported_InAttributes(
+                SyntaxConstants.CSharp.LayoutKeyword));
+        }
+
+        /// <summary>
+        /// Writes code for the given <paramref name="chunk"/>.
+        /// </summary>
+        /// <param name="chunk">The <see cref="StatementChunk"/> to render.</param>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown unconditionally to inform user of unexpected code block.
+        /// </exception>
+        protected override void Visit(StatementChunk chunk)
+        {
+            throw new InvalidOperationException(RazorResources.TagHelpers_CodeBlocks_NotSupported_InAttributes);
+        }
+
+        /// <summary>
+        /// Writes code for the given <paramref name="chunk"/>.
+        /// </summary>
+        /// <param name="chunk">The <see cref="TemplateChunk"/> to render.</param>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown unconditionally to inform user of unexpected template e.g. <c>@&lt;p&gt;paragraph@&lt;/p&gt;</c>.
+        /// </exception>
+        protected override void Visit(TemplateChunk chunk)
+        {
+            throw new InvalidOperationException(
+                RazorResources.FormatTagHelpers_InlineMarkupBlocks_NotSupported_InAttributes(_attributeTypeName));
+        }
+
+        // Tracks the code mapping and writes code for a leaf node in the attribute value Chunk tree.
+        private void RenderCode(string code, Span association)
+        {
+            if (firstChild)
+            {
+                firstChild = false;
+
+                // Indent to match placement within line whether or not line starts with whitespace. Since last action
+                // was to write out line pragma with a final newline, Indent() call always indents.
+                Writer.Indent(association.Start.CharacterIndex);
```

Comment:
I'll try again but it previously seemed to consistently return an empty string in tested scenarios...